### PR TITLE
Add templates for story_cloze

### PR DESCRIPTION
**Pull request by zaidalyafeai**
Added a custom data loader script for the offline dataset `story_cloze`. Also, created a candidate template.

### CODE REVIEW

**Comment by VictorSanh** [comment]
Thanks for opening this PR @zaidalyafeai !

I was initially suggesting that:
- your `story_cloze.py` goes to huggingface/datasets via a PR (as a dataset that you need manual download)
- the datasets library version is installed locally from your PR (in your local version, you would specify the local path where you need to load the data i.e. `path_to_manual_folder`)
- Ultimately, we would probably have a "data_dir" box in the promptsource to specify the data_dir in the tool rather than specifying it in the code, but we don't need that now since the priority is to get the eval suite ready

This has multiple advantages:
- you would still be able to prompt this dataset in promptsource
- the things that should to be in `datasets` (storycloze.py, datasets_info.json) would be in `datasets` in not in promptsource
- when we do seqio caching, that would work out of the box -> right now I don't think that works

btw, you shouldn't push the `.lock` file (even though I believe in hf datasets, it's flagged in the `.gitignore`)

**Comment by zaidalyafeai** [comment]
Thanks @VictorSanh, I already opened a [PR ](https://github.com/huggingface/datasets/pull/2907)to add it to huggingface datasets but it seems because of the large number of open PRs it will take sometime to revise and merge. I was suggesting this approach in case if we need to add custom datasets quickly so that we don't need to wait for PRs to be merged to `datasets`. Since @awebson is saying that this dataset is a must-have, what do you suggest is the next thing to do ?

**Comment by VictorSanh** [comment]
> Thanks @VictorSanh, I already opened a [PR ](https://github.com/huggingface/datasets/pull/2907)to add it to huggingface datasets but it seems because of the large number of open PRs it will take sometime to revise and merge. I was suggesting this approach in case if we need to add custom datasets quickly so that we don't need to wait for PRs to be merged to `datasets`. Since @awebson is saying that this dataset is a must-have, what do you suggest is the next thing to do ?

Oh awesome, I didn't find it initially for some reason...

Yes, please do:
- the datasets library version is installed locally from your PR (in your local version, you would specify the local path where you need to load the data i.e. path_to_manual_folder)
- remove the three files (.lock, story_cloze.py and datasets_info.json)
- remove CUSTOM_DATASETS related stuff in `promptsource.py`

once this is done, i will ask you to seqio cache the dataset and upload it to the hub (i will share the command with you later)

thank you!

**Comment by zaidalyafeai** [comment]
Sounds good, what about the templates, for `story_cloze`, do I add them as part of this PR or just discard it?

**Comment by VictorSanh** [comment]
yes, let's have the template here in this pr!

**Comment by zaidalyafeai** [comment]
@VictorSanh I followed your steps and this seems to work fine. I want to test the seqio cache script early on some test templates to make sure it works fine.

**Comment by zaidalyafeai** [comment]
Since the dataset requires manual download and is not part of hf datasets maybe it should be excluded from the build checks?

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+
+      - {{answer_choices | join("\n - ")}}
+
+      |||
+
+
+      {{answer_choices[answer_right_ending -1]}}'
+    metadata: !TemplateMetadata
+      choices_in_prompt: true
+      metrics: []
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:27]:
```suggestion
      metrics:
      - Accuracy
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+      metrics:
+      - Accuracy
+      original_task: true
+    name: Choose Story Ending
+    reference: ''
+  9dab69d1-cad0-4d2f-a7cc-120df233571c: !Template
+    answer_choices: null
+    answer_choices_key: '{{sentence_quiz1}} ||| {{sentence_quiz2}}'
+    id: 9dab69d1-cad0-4d2f-a7cc-120df233571c
+    jinja: 'Yesterday, I watched a movie. These plots happened :
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:50]:
```suggestion
    jinja: 'Yesterday, I watched a movie. Here's what happened:
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+      {{input_sentence_1}}
+
+      {{input_sentence_2}}
+
+      {{input_sentence_3}}
+
+      {{input_sentence_4}}
+
+
+      What happens next ?
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:62]:
```suggestion
      What happens next?
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+      - BLEU
+      - ROUGE
+      original_task: false
+    name: Generate Ending
+    reference: ''
+  c6f3d802-4f97-449f-a911-03470d418f7d: !Template
+    answer_choices: null
+    answer_choices_key: '{{sentence_quiz1}} ||| {{sentence_quiz2}}'
+    id: c6f3d802-4f97-449f-a911-03470d418f7d
+    jinja: 'I read a the following novel:
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:125]:
```suggestion
    jinja: 'I read the following novel:
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+
+      {{input_sentence_2}}
+
+      {{input_sentence_3}}
+
+      {{input_sentence_4}}
+
+
+      What do you think is the most probably ending ? You can choose from the following
+      options :
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:138]:
```suggestion
      What do you think is the most probable ending? You can choose from the following
      options:
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+
+
+      - {{answer_choices | join("\n - ")}}
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:141]:
```suggestion
      - {{answer_choices | join("\n- ")}}
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+      {{input_sentence_2}}
+
+      {{input_sentence_3}}
+
+      {{input_sentence_4}}
+
+
+      What is a possible continuation for the story given the following options ?
+
+      - {{answer_choices | join("\n - ")}}
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:19]:
```suggestion
      - {{answer_choices | join("\n- ")}}
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+
+      {{input_sentence_3}}
+
+      {{input_sentence_4}}
+
+
+      What happens next ?
+
+
+      - {{answer_choices | join("\n - ")}}
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:65]:
```suggestion
      - {{answer_choices | join("\n- ")}}
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 9]
```diff
@@ -0,0 +1,152 @@
...
+      original_task: true
+    name: Movie What Happens Next
+    reference: ''
+  b5c8445f-2d3a-4691-bdd5-58956816702f: !Template
+    answer_choices: null
+    answer_choices_key: '{{sentence_quiz1}} ||| {{sentence_quiz2}}'
+    id: b5c8445f-2d3a-4691-bdd5-58956816702f
+    jinja: "What is a possible continuation for the following story ? \n\n{{input_sentence_1}}\n\
+      {{input_sentence_2}}\n{{input_sentence_3}}\n{{input_sentence_4}}\n\nChoose from\
+      \ the following options:\n- {{answer_choices | join(\"\\n - \")}}\n|||\n\n{{answer_choices[answer_right_ending\
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:84]:
```suggestion
      \ the following options:\n- {{answer_choices | join(\"\\n- \")}}\n|||\n\n{{answer_choices[answer_right_ending\
```

File: promptsource/templates/story_cloze/2016/templates.yaml [hunk 10]
```diff
@@ -0,0 +1,152 @@
...
+    name: Answer Given options
+    reference: ''
+  1a9d53bc-eb77-4e7c-af6e-3d15b79d6cf1: !Template
+    answer_choices: null
+    answer_choices_key: '{{sentence_quiz1}} ||| {{sentence_quiz2}}'
+    id: 1a9d53bc-eb77-4e7c-af6e-3d15b79d6cf1
+    jinja: "Read the following story :\n\n{{input_sentence_1}}\n{{input_sentence_2}}\n\
+      {{input_sentence_3}}\n{{input_sentence_4}}\n\nChoose a possible ending for the\
+      \ previous story from the following options: \n- {{answer_choices | join(\"\\\
+      n - \")}}\n|||\n\n{{answer_choices[answer_right_ending -1]}}"
```
Comment (VictorSanh) [inline promptsource/templates/story_cloze/2016/templates.yaml:38]:
```suggestion
      n- \")}}\n|||\n\n{{answer_choices[answer_right_ending -1]}}"
```

**Review by VictorSanh — APPROVED** [review]
from what i can saw from the data, LGTM! corrected a few minor things.

Letting @awebson validating one more time and we can cache it.

Note: I see you only did the 2016 subset. I see the recommend evaluation subset is 2018, is that a problem? (FLAN report numbers on the 2016 subset)

File: promptsource/templates/story_cloze/2016/templates.yaml (hunk 3 above)
Comment (awebson) [reply promptsource/templates/story_cloze/2016/templates.yaml:62]:
```suggestion
      What happened next ?
```

**Comment by zaidalyafeai** [comment]
I discussed that with @awebson, the 2018 doesn't have labels for the test split only for validation. We can copy the same templates to the 2018 validation if you want to include that. Between in GPT-3 the use the 2016 subset.

**Pull request merged**